query/iterator_unittest.cu

```diff
@@ -504,8 +503,7 @@ TEST(MeasureIteratorTest, CheckMax) {
 // cppcheck-suppress *
 TEST(RecordIDJoinIteratorTest, CheckIterator) {
   RecordID recordIDsH[5];
-  thrust::host_vector<ForeignTableIterator<int32_t>> vpItersHost(5);
-  ForeignTableIterator<int32_t> *vpIters = vpItersHost.data();
+  ForeignTableIterator<int32_t> vpItersH[5];
```

Comment:
are these test changes related to the cmake change?

Reply:
Some of the test changes are, like the bugs we have seen last time where nvcc hangs when compilation. 
Some of the test changes are just refactoring the code